answer.go

```diff
@@ -34,11 +33,17 @@ type Promise struct {
 	//	  Next state is resolved.
 	//	- Resolved.  Fulfill or Reject has finished.
 
-	// mu protects the fields below.  When acquiring multiple Promise.mu
-	// mutexes, they must be acquired in traversal order (i.e. p, then
-	// p.next, then p.next.next).
-	mu sync.Mutex
+	// When acquiring multiple Promise.state mutexes, they must be acquired
+	// in traversal order (i.e. p, then p.next, then p.next.next).
+	//
+	// FIXME: is the above traversal scheme sound? It depends on avoiding
```

Comment:
Note: this is an existing problem, though I'm not sure whether we actually rely on this in the code anymore or if it's just the comment that's bitrotten -- I have vague memories of fixing a bug that was caused by this bad assumption, so we may be good.

Anyway, this is mostly unrelated, I just spotted it while making the change.